aderyn_core/src/detect/low/uninitialized_state_variable.rs

```diff
@@ -0,0 +1,150 @@
+use std::collections::{BTreeMap, HashSet};
+use std::error::Error;
+
+use crate::ast::{Expression, NodeID};
+
+use crate::capture;
+use crate::detect::detector::IssueDetectorNamePool;
+use crate::{
+    context::workspace_context::WorkspaceContext,
+    detect::detector::{IssueDetector, IssueSeverity},
+};
+use eyre::Result;
+
+#[derive(Default)]
+pub struct UninitializedStateVariableDetector {
+    // Keys are: [0] source file name, [1] line number, [2] character location of node.
+    // Do not add items manually, use `capture!` to add nodes to this BTreeMap.
+    found_instances: BTreeMap<(String, usize, String), NodeID>,
+}
+
+impl IssueDetector for UninitializedStateVariableDetector {
+    fn detect(&mut self, context: &WorkspaceContext) -> Result<bool, Box<dyn Error>> {
+        /*
```

Comment:
I like this plan! It sounds like this will correctly _not_ capture state variables that are declared, but initialized in a contract that extends the contract the variable was declared. For example:

```sol

contract base{
    uint256 public myVar;
}

contract extension is base {
    constructor() {
        myVar = 4;
    }
}
```

Reply:
In fact, could you add a test for this please @TilakMaddy ?

Reply:
Test worked! It has correctly not captured it :)

Reply:
✅